Add Cerberus to QMK

awkannan opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

Add Cerberus PCBs to QMK

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

zvecr commented on keyboards/cannonkeys/cerberus/readme.md:
```diff
@@ -0,0 +1,24 @@
+
+This is used for both the Solderable and Hotswap version of Cerberus
+
+The hotswap PCB does not support per-key LEDs
```
Same as <https://github.com/qmk/qmk_firmware/pull/24227#discussion_r1700724652>

dunk2k commented on keyboards/cannonkeys/cerberus/keyboard.json:
```diff
@@ -0,0 +1,755 @@
+        "pin": "B15"
+    },
+    "community_layouts": ["tkl_nofrow_ansi", "tkl_nofrow_iso"],
+    "layout_aliases": {
+        "LAYOUT_all": "LAYOUT"
+    },
+    "layouts": {
+        "LAYOUT": {
```
layout alias not required as renamed `LAYOUT` to `LAYOUT_all`
```suggestion
    "layouts": {
        "LAYOUT_all": {
```

waffle87 commented on keyboards/cannonkeys/cerberus/hotswap/keyboard.json:
```diff
@@ -0,0 +1,314 @@
+{
+    "keyboard_name": "Cerberus HS",
+    "features": {
+        "backlight": false
+    },
```
```suggestion
```

drashna commented on keyboards/cannonkeys/cerberus/solderable/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,25 @@
+};
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [_BASE] = LAYOUT_all(
+		KC_ESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSPC, KC_GRV, KC_INS, KC_HOME, KC_PGUP,
+		KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL, KC_END, KC_PGDN,
+		KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_NUHS, KC_ENT,
+		KC_LSFT, KC_NUBS, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_LSFT, MO(1), KC_UP,
+		KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, KC_LALT, MO(1), KC_LGUI, KC_LCTL, KC_LEFT, KC_DOWN, KC_RGHT
```
```suggestion
        KC_ESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSPC, KC_GRV, KC_INS, KC_HOME, KC_PGUP,
        KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL, KC_END, KC_PGDN,
        KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_NUHS, KC_ENT,
        KC_LSFT, KC_NUBS, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_LSFT, MO(1), KC_UP,
        KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, KC_LALT, MO(1), KC_LGUI, KC_LCTL, KC_LEFT, KC_DOWN, KC_RGHT
```

drashna reviewed (changes requested):
As of August 26, 2024, `qmk/qmk_firmware` is no longer accepting VIA-enabled keymaps as these have now transitioned to a repository under the VIA team's control.

As you've submitted a PR containing `via` or VIA-enabled keymap(s), this is your notice that they should be removed from this PR. You should now submit a secondary PR to the [VIA QMK Userspace](https://github.com/the-via/qmk_userspace_via) repository with your associated `via` or VIA-enabled keymaps instead.

awkannan commented:
@drashna Ready on this one too

fauxpark commented on keyboards/cannonkeys/cerberus/info.json:
```diff
@@ -0,0 +1,50 @@
+            "knight": true,
+            "rainbow_mood": true,
+            "rainbow_swirl": true,
+            "rgb_test": true,
```
```suggestion
```

fauxpark replied:
Still outstanding.

fauxpark commented on keyboards/cannonkeys/cerberus/solderable/keymaps/default/keymap.c:
RGBLight enabled but no `UG_` keycodes?

fauxpark commented on keyboards/cannonkeys/cerberus/hotswap/keymaps/default/keymap.c:
As above.

awkannan commented:
@fauxpark this one had the opposite problem compared to the others haha

I've fixed it and added in the RGB control keycodes.

Thanks! Ready for further review.

fauxpark commented on keyboards/cannonkeys/cerberus/solderable/keyboard.json:
```diff
@@ -0,0 +1,711 @@
+{
+    "keyboard_name": "Cerberus",
+    "backlight": {
+        "breathing": true,
+        "levels": 6,
+        "pin": "A6"
+    },
+    "features": {
+        "backlight": true
+    },
```
Backlight config but no keycodes

awkannan replied:
Just took care of this - thanks again for the catch.

Ready for review again.

waffle87 commented on keyboards/cannonkeys/cerberus/readme.md:
```diff
@@ -0,0 +1,22 @@
+* Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)
+* Hardware Supported: STM32F072CBT6
+
+Make example for this keyboard (after setting up your build environment):
+
+    make cannonkeys/cerberus:solderable:default
```
```suggestion
Make example for this keyboard (after setting up your build environment):

    make cannonkeys/cerberus/hotswap:default
    make cannonkeys/cerberus/solderable:default
    
Flashing example for this keyboard:

    make cannonkeys/cerberus/hotswap:default:flash
    make cannonkeys/cerberus/solderable:default:flash
```
